lib/block/blockdownload.js

```diff
@@ -112,40 +89,127 @@ class BlockDownload extends EventEmitter {
+     * and file (download) error.
+     * @returns {TransferController} Transfer controller
+     */
+    createBlockDownloadController() {
+        const controller = new TransferController();
+        this.errorEvents = [];
+
+        controller.on(TransferEvents.CREATE_TRANSFER_PARTS, transferEvent => {
+            this.emit("filestart", transferEvent.transferAsset.eventData);
+        });
+
+        controller.on(TransferEvents.JOIN_TRANSFER_PARTS, transferEvent => {
+            this.emit("fileprogress", {
+                ...transferEvent.transferAsset.eventData,
+                transferred: transferEvent.props.transferBytes
+            });
+        });
+
+        controller.on(TransferEvents.AFTER_JOIN_TRANSFER_PARTS, transferEvent => {
+            this.emit("fileend", transferEvent.transferAsset.eventData);
+        });
+
+        controller.on(TransferEvents.ERROR, transferEvent => {
+            console.log(`Error during block download: ${transferEvent.error}`);
+            this.errorEvents.push(transferEvent);
+
+            if (transferEvent.props.firstError) {
+                this.emit("fileerror", {
+                    ...transferEvent.transferAsset.eventData,
+                    errors: [DownloadError.fromError(transferEvent.error)]
+                });
+            }
+        });
+
+        console.log("Block download: block download controller created");
+        return controller;
+    }
+
+    /**
+     * Removes all listeners for events 
+     * `CREATE_TRANSFER_PARTS`, `JOIN_TRANSFER_PARTS`,
+     * `AFTER_JOIN_TRANSFER_PARTS` and `ERROR`
+     * from a block download controller
+     * @param {TransferController} controller a block download controller
+     */
+    finalizeController(controller){
+        if(!controller){
+            return;
+        }
+        
+        try {
+            controller.removeAllListeners(TransferEvents.CREATE_TRANSFER_PARTS);
+            controller.removeAllListeners(TransferEvents.JOIN_TRANSFER_PARTS);
+            controller.removeAllListeners(TransferEvents.AFTER_JOIN_TRANSFER_PARTS);
+            controller.removeAllListeners(TransferEvents.ERROR);
+        } catch (err) {
+            console.log(`Failed to remove event listeners from block download controller: ${err}`);
+        }
+    }
+
     /**
      * Download files to local disk
      * Throws the first unrecoverable error if unsuccessful, all others are logged
      * 
      * @param {BlockDownloadOptions} options Block download options
      */
-    async downloadFiles (options = {}) {
+    async downloadFiles(options = {}) {
         const preferredPartSize = options.preferredPartSize || DEFAULT_PART_SIZE;
         const maxConcurrent = options.maxConcurrent || DEFAULT_MAX_CONCURRENCY;
-        const controller = createController();
 
         const retryOptions = {
-            retryMaxCount: options.retryCount || DEFAULT_RETRY_COUNT
+            retryMaxCount: options.retryCount || DEFAULT_RETRY_COUNT,
```

Comment:
retryMaxCount is a tricky one...it actually overrides `maxRetryDuration` if it is set: https://github.com/adobe/node-httptransfer/blob/6e588e161cb47541d5b5d13aa5a2db6522631e02/lib/retry.js#L22
since you have a default value for retryCount, this would mean a client passing in `retryMaxDuration` would not be applied.

for this reason, I removed`DEFAULT_RETRY_COUNT` from blockupload. Additionally, we had to do this for upload since we pass in `retryMaxDuration` in the pipeline and this would break the existing retry functionality: https://github.com/adobe/asset-compute-pipeline/blob/master/lib/storage/http.js#L66
(we retry for 15mins for multipart upload instead of the default 1 min)

I would suggest two changes (this is what I did in blockupload too):
1. remove default retry count
2. remove new api/name for retryCount: `retryCount` and use the previous name: `retryMaxCount`